python/astro_metadata_translator/translators/decam.py

```diff
@@ -331,6 +331,12 @@ def read_all_headers(cls, filename, primary=None):
         # Circular dependency so must defer import
         from ..headers import merge_headers
 
+        # If this is not a FITS file return the primary if defined.
+        # It is likely that the metadata came from a test YAML file.
+        if not re.search(r"\.fit[s]?\b", filename) and primary is not None:
+            yield primary
+            return
+
```

Comment:
I've updated the documentation but now that I think about it it might be better to have `astrometadata translate` skip over this call completely if a test YAML file is encountered.